plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestCassandraConnectorTest.java

```diff
@@ -237,6 +238,32 @@ public void testCharVarcharComparison()
                 .hasMessage("unsupported type: char(3)");
     }
 
+    @Test
+    public void testAllPartitionKeyTypesPushdownPredicate()
```

Comment:
`TestCassandraConnectorTest` tests on Cassandra `2.2`.
However the code has the following logic

https://github.com/trinodb/trino/blob/master/plugin/trino-cassandra/src/main/java/io/trino/plugin/cassandra/CassandraSession.java#L369-L376

for retrieving the partitions

https://github.com/trinodb/trino/blob/master/plugin/trino-cassandra/src/main/java/io/trino/plugin/cassandra/CassandraSession.java#L369-L376

Any advice on how could we cover the usecase of dealing with Cassandra 2.2+ ?

Reply:
Not sure if I understood your question correctly, but the image name `2.2` gets the latest `2.2.x` (=2.2.19 today). The comparison (2.2.19 compareTo 2.2 > 0) will return `true`.

Reply:
Sorry for the confusion.

I was wondering how do we get to test the `else` branch.

```
            log.debug("Using combination of partition values to fetch partitions.");
            rows = queryPartitionKeysLegacyWithMultipleQueries(table, filterPrefixes);
```

Reply:
We can get the coverage by running tests on Cassandra `2.1`. It will require some refactoring of tests though.